### PR TITLE
Update extractor_callbacks.cpp

### PR DESCRIPTION
**Pull request by mostafa-j13**
way_id overflow with 64 bit number

# Issue

https://github.com/Project-OSRM/osrm-backend/issues/5380

### CODE REVIEW

File: src/extractor/extractor_callbacks.cpp [hunk 1]
```diff
@@ -478,7 +478,7 @@ void ExtractorCallbacks::ProcessWay(const osmium::Way &input_way, const Extracti
                    });
 
     external_memory.way_start_end_id_list.push_back(
-        {OSMWayID{static_cast<std::uint32_t>(input_way.id())},
```
Comment (mostafa-j13) [inline src/extractor/extractor_callbacks.cpp]: here this must be uint64_t
 #5380

Comment (mostafa-j13) [reply src/extractor/extractor_callbacks.cpp]: i fixed in #5381

**Review by akashihi — APPROVED** [review]

**Review by akashihi — CHANGES_REQUESTED** [review]
I think the issue it with 33bit node ids, not way ids?

**Comment by wangyoucao577** [comment]
> I think the issue it with 33bit node ids, not way ids?

It's indeed `way_id` overflow, not only `node_id`. But it's mostly for non-OSM data set since the biggest OSM `way_id` value is about 565140676, which fits within 2^30. See comments in https://github.com/Project-OSRM/osrm-backend/issues/5380#issuecomment-468330377

**Comment by SiarheiFedartsou** [comment]
done in https://github.com/Telenav/osrm-backend/commit/6283c6074066f98e6d4a9f774f21ea45407c0d52

**Pull request closed without merging**